npm error due to package-lock.json when running any script

<!--
  - Thanks for taking the time to report a bug in the Oppia project.
  - Before filing a new issue, please do a quick search to check that it hasn't
  - already been filed on the [issue tracker](https://github.com/oppia/oppia/issues)._
  -->

**Describe the bug**
When I run any script, including start.sh, backend-test, frontend-test, etc, the process would be stopped due to npm errors. More discussion in https://github.com/oppia/oppia/pull/6913
<img width="680" alt="Screen Shot 2019-06-12 at 11 04 27 AM" src="https://user-images.githubusercontent.com/32754302/59415306-10441500-8d78-11e9-9bfc-2526a2a14108.png">
It could be solved by deleting package-lock.json and node_modules, and run start.sh again.

**To Reproduce**
Steps to reproduce the behavior:
1. Run start.sh

**Observed behavior**
Local stack won't be started due to npm errors.

**Expected behavior**
Local stack should be started.


**Desktop (please complete the following information; delete this section if the issue does not arise on desktop):**
 - OS: MacOS 10.14.5

So I just tried this on my machine. The first time I pulled from develop and ran start.sh it gave me a bunch of similar npm ERR!, but after I Ctrl-C'd that and ran start.sh again it passed without issues (and gave no errors). Weirdly enough, package-lock.json was not modified.

@sophiewu6 do you get this blocking error on develop after syncing to HEAD? Also, there was probably a commit sometime ago where you did not get this error log, right? If so, can you use bisection to figure out which commit caused the regression in the setup process?

--- a/scripts/setup.sh
+++ b/scripts/setup.sh
@@ -246,11 +246,4 @@ if ! test_python_version $PYTHON_CMD; then
 fi
 export PYTHON_CMD
 
-# List all node modules that are currently installed. The "npm list" command is
-# slow, so we precompute this here and refer to it as needed.
-echo "Generating list of installed node modules..."
-NPM_INSTALLED_MODULES="$($NPM_CMD list)"
-export NPM_INSTALLED_MODULES
-echo "Generation completed."
-
 export SETUP_DONE=true

